src/hdmf/spec/namespace.py

```diff
@@ -229,13 +229,28 @@ class NamespaceCatalog:
             {'name': 'dataset_spec_cls', 'type': type,
              'doc': 'the class to use for dataset specifications', 'default': DatasetSpec},
             {'name': 'spec_namespace_cls', 'type': type,
-             'doc': 'the class to use for specification namespaces', 'default': SpecNamespace})
+             'doc': 'the class to use for specification namespaces', 'default': SpecNamespace},
+            {'name': 'core_namespace', 'type': str,
+             'doc': 'the name of the core namespace', 'default': None})
     def __init__(self, **kwargs):
         """Create a catalog for storing  multiple Namespaces"""
         self.__namespaces = OrderedDict()
         self.__dataset_spec_cls = getargs('dataset_spec_cls', kwargs)
         self.__group_spec_cls = getargs('group_spec_cls', kwargs)
         self.__spec_namespace_cls = getargs('spec_namespace_cls', kwargs)
+
+        core_namespace = getargs('core_namespace', kwargs)
+        if core_namespace is None:
+            from ..common import CORE_NAMESPACE, EXP_NAMESPACE
+            # NOTE: even though HDMF does not guarantee backwards compatibility with schema
+            # using an older version of the experimental namespace, in practice, this has not been
+            # an issue, and it is costly to determine whether there is an incompatibility before issuing
+            # a warning. so, we ignore the experimental namespace warning by default.
+            # see https://github.com/hdmf-dev/hdmf/pull/1258
+            self.__core_namespace = [CORE_NAMESPACE, EXP_NAMESPACE]
```

Comment:
I think this should be handled in https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/common/__init__.py#L236

This class should not know anything about hdmf.common.

To make this work, I think the `core_namespace` string argument should change to `core_namespaces` as a list of strings.